File: lab3/backend/products/soa-products-ejb/src/main/kotlin/com/soa/products/ejb/domain/Product.kt
package com.soa.products.ejb.domain

import java.io.Serializable
import java.time.LocalDate

data class Product(
    val id: Long,
    val name: String,
    val coordinates: Coordinates,
    val creationDate: LocalDate,
    val price: Long,
    val partNumber: String?,
    val manufacturerCost: Long,
    val unitOfMeasure: UnitOfMeasure,
    val owner: Person?,
) : Serializable

data class Coordinates(
    val x: Int,
    val y: Int,
) : Serializable

enum class UnitOfMeasure {
    KILOGRAMS,
    METERS,
    MILLILITERS,
    MILLIGRAMS,
}

File: lab4/frontend/libs/shared/src/lib/shared/models/product.ts
import { Coordinates } from "./coordinates";
import { Owner } from "./owner";
import { UnitOfMeasureType } from "./unit-of-measure.enum";

export interface Product {
    price: number;
    name: string;
    coordinates: Coordinates;
    manufacturerCost: number;
    partNumber: string;
    id: number;
    owner: Owner;
    createdDate: string;
    unitOfMeasure: UnitOfMeasureType;
}